src/garage/torch/algos/sac.py

```diff
@@ -0,0 +1,258 @@
+                 qf1,
+                 qf2,
+                 target_qf1,
+                 target_qf2,
+                 replay_buffer,
+                 alpha=None,
+                 gradient_steps_per_itr=1000,
+                 target_entropy=None,
+                 initial_log_entropy=0.,
+                 use_automatic_entropy_tuning=False,
+                 discount=0.99,
+                 max_path_length=None,
+                 buffer_batch_size=64,
+                 min_buffer_size=int(1e4),
+                 target_update_tau=5e-3,
+                 policy_lr=3e-4,
+                 qf_lr=3e-4,
+                 optimizer=torch.optim.Adam,
+                 clip_pos_returns=False,
+                 clip_return=np.inf,
+                 max_action=None,
+                 smooth_return=True,
+                 input_include_goal=False):
+
+        self.policy = policy
+        self.qf1 = qf1
+        self.qf2 = qf2
+        self.replay_buffer = replay_buffer
+        self.tau = target_update_tau
+        self.policy_lr = policy_lr
+        self.qf_lr = qf_lr
+        self.initial_log_entropy = initial_log_entropy
+        self.gradient_steps = gradient_steps_per_itr
+        self.clip_pos_returns = clip_pos_returns
+        self.clip_return = clip_return
+        self.evaluate = False
+        self.input_include_goal = input_include_goal
+
+        super().__init__(env_spec=env_spec,
+                         policy=policy,
+                         qf=qf1,
+                         n_train_steps=self.gradient_steps,
+                         n_epoch_cycles=1,
+                         max_path_length=max_path_length,
+                         buffer_batch_size=buffer_batch_size,
+                         min_buffer_size=min_buffer_size,
+                         replay_buffer=replay_buffer,
+                         use_target=True,
+                         discount=discount,
+                         smooth_return=smooth_return)
+
+        # use 2 target q networks
+        self.target_qf1 = copy.deepcopy(self.qf1)
+        self.target_qf2 = copy.deepcopy(self.qf2)
+        # self.target_qf1 = target_qf1
+        # self.target_qf2 = target_qf2
+        self.policy_optimizer = optimizer(self.policy.parameters(),
+                                          lr=self.policy_lr)
+        self.qf1_optimizer = optimizer(self.qf1.parameters(), lr=self.qf_lr)
+        self.qf2_optimizer = optimizer(self.qf2.parameters(), lr=self.qf_lr)
+        # automatic entropy coefficient tuning
+        self.use_automatic_entropy_tuning = use_automatic_entropy_tuning
+        if self.use_automatic_entropy_tuning and not alpha:
+            if target_entropy:
+                self.target_entropy = target_entropy
+            else:
+                self.target_entropy = -np.prod(
+                    self.env_spec.action_space.shape).item()
+            self.log_alpha = torch.tensor([self.initial_log_entropy], dtype=torch.float, requires_grad=True)
```

Comment:
Probably in case someone sets `initial_log_entropy` to `0` (an int). It's a good idea to do this, since otherwise the user will get a cryptic error in the middle of SAC. Maybe add a comment why though?